sdk/sdk-api/src/main/java/mil/emp3/api/MilStdSymbol.java

```diff
@@ -102,61 +102,9 @@ public String toString() {
          * @return affiliation enumerated value or null if the string does not represent a valid MilStd affiliation code.
          */
         public static Affiliation fromString(String str) {
-            Affiliation eValue = null;
-
-            if ((str == null) || (str.length() < 1)) {
-                return eValue;
-            }
-
-            switch (str.substring(0, 1)) {
-                case "P":
-                    eValue = PENDING;
-                    break;
-                case "U":
-                    eValue = UNKNOWN;
-                    break;
-                case "F":
-                    eValue = FRIEND;
-                    break;
-                case "N":
-                    eValue = NEUTRAL;
-                    break;
-                case "H":
-                    eValue = HOSTILE;
-                    break;
-                case "A":
-                    eValue = ASSUMED_FRIEND;
-                    break;
-                case "S":
-                    eValue = SUSPECT;
-                    break;
-                case "G":
-                    eValue = EXERCISE_PENDING;
-                    break;
-                case "W":
-                    eValue = EXERCISE_UNKNOWN;
-                    break;
-                case "D":
-                    eValue = EXERCISE_FRIEND;
-                    break;
-                case "L":
-                    eValue = EXERCISE_NEUTRAL;
-                    break;
-                case "M":
-                    eValue = EXERCISE_ASSUMED_FRIEND;
-                    break;
-                case "J":
-                    eValue = JOKER;
-                    break;
-                case "K":
-                    eValue = FAKER;
-                    break;
-                default:
-                    eValue = FRIEND;
-                    break;
-            }
-
-            return eValue;
+            if (str == null || str.isEmpty())
+                return null;
```

Comment:
Did we ever decide if it was better to throw an exception versus returning null?  I remember this being discussed on the last PR.  I would always lean on throwing an exception but I would be open to hear why you chose to return null instead.

Reply:
The null is checked for in places where it is called.  It is a public method and I don't want to change the interface.   It is more expensive to throw an exception than check for a null.

Reply:
fair enough